resources/asciidoctor/lib/edit_me/extension.rb

```diff
@@ -28,50 +28,73 @@ def process(document)
       url = url[0..-2] if url.end_with? '/'
       edit_urls << { toplevel: toplevel, url: url }
     end
-    edit_urls.reverse!
+    # Prefer the longest matching edit url
+    edit_urls = edit_urls.sort_by { |e| [-e[:toplevel].length, e[:toplevel]] }
     document.attributes['edit_urls'] = edit_urls
     super
   end
 
   def process_block(block)
     return unless %i[preamble section floating_title].include? block.context
 
-    def block.title
+    block.extend WithEditLink
+    if block.context == :preamble
+      def block.source_path
+        # source_location.path doesn't work for relative includes outside of
+        # the base_dir which we use when we build books from many repos.
+        document.source_location.file
+      end
+    else
+      def block.source_path
+        # source_location.path doesn't work for relative includes outside of
+        # the base_dir which we use when we build books from many repos.
+        source_location.file
+      end
+    end
+  end
+
+  ##
+  # Extension to blocks that need an "edit me" link.
+  module WithEditLink
+    def title
+      url = edit_url
```

Comment:
Is there a reason you set the `url` variable here?
You can do `return super unless edit_url`
Then use the `edit_url` in the string as you've done in line 63.